src/e2e/java/teammates/e2e/cases/lnp/StudentUpdateLNPTest.java

```diff
@@ -0,0 +1,331 @@
+
+                return headers;
+            }
+
+            @Override
+            public List<List<String>> generateCsvData() {
+                DataBundle dataBundle = loadDataBundle(getJsonDataPath());
+                List<List<String>> csvData = new ArrayList<>();
+
+                dataBundle.instructors.forEach((key, instructor) -> {
+                    List<String> csvRow = new ArrayList<>();
+
+                    csvRow.add(instructor.googleId);
+                    csvRow.add("no");
+                    csvRow.add(COURSE_ID);
+                    csvRow.add(STUDENT_ID);
+                    csvRow.add(STUDENT_EMAIL);
+                    csvRow.add(FEEDBACK_SESSION_NAME);
+                    csvRow.add(QUESTION_ID);
+
+                    for (int i = 1; i <= NUMBER_OF_FEEDBACK_RESPONSES; i++) {
+                        csvRow.add(FEEDBACK_RESPONSE_ID + " " + i);
+                    }
+
+                    StudentUpdateRequest studentUpdateRequest = new StudentUpdateRequest(
+                            UPDATE_STUDENT_NAME,
+                            UPDATE_STUDENT_EMAIL,
+                            TEAM_NAME,
+                            GIVER_SECTION_NAME,
+                            STUDENT_COMMENTS,
+                            false
+                    );
+
+                    String updateData = sanitizeForCsv(JsonUtils.toJson(studentUpdateRequest));
+                    csvRow.add(updateData);
+
+                    csvData.add(csvRow);
+                });
+
+                return csvData;
+            }
+        };
+    }
+
+    private Map<String, String> getRequestHeaders() {
+        Map<String, String> headers = new HashMap<>();
+
+        headers.put("X-CSRF-TOKEN", "${csrfToken}");
+        headers.put("Content-Type", "application/json");
+
+        return headers;
+    }
+
+    private String getTestEndpoint() {
+        return Const.ResourceURIs.URI_PREFIX + Const.ResourceURIs.STUDENT
+            + "?courseid=${courseId}&studentid=${studentId}&studentemail=${studentEmail}";
+    }
+
+    @Override
+    protected ListedHashTree getLnpTestPlan() {
+        ListedHashTree testPlan = new ListedHashTree(JMeterElements.testPlan());
+        HashTree threadGroup = testPlan.add(
+                JMeterElements.threadGroup(NUM_INSTRUCTORS, RAMP_UP_PERIOD, 1));
+
+        threadGroup.add(JMeterElements.csvDataSet(getPathToTestDataFile(getCsvConfigPath())));
+        threadGroup.add(JMeterElements.cookieManager());
+        threadGroup.add(JMeterElements.defaultSampler());
+
+        threadGroup.add(JMeterElements.onceOnlyController())
+            .add(JMeterElements.loginSampler())
+            .add(JMeterElements.csrfExtractor("csrfToken"));
+
+        // Add HTTP sampler for test endpoint
+        HeaderManager headerManager = JMeterElements.headerManager(getRequestHeaders());
+        threadGroup.add(JMeterElements.httpSampler(getTestEndpoint(), PUT, "${updateData}"))
+            .add(headerManager);
```

Comment:
The lint errors in the CI are for the indentation in 293/4 & 299. You can try running `./gradlew lint` locally to catch them :)